### DISCUSSION TITLE
Will you make an android port?

### DISCUSSION THREAD

**Discussion by ClixzyFN** [nzp-team/nzportable#984] [General]
It would be cool to have a android port of nzp on mobile i mean it's in the name

**Comment by MotoLegacy** [comment]
Not looking to extend the projects port porfolio

**Reply by BrownNPC** [reply]
how hard would it be to compile fteqw for android? Or maybe adding touch support to the web version?

**Reply by Peter0x44** [reply]
How hard it is doesn't matter, no one is going to do it for you.

**Reply by BrownNPC** [reply]
> How hard it is doesn't matter, no one is going to do it for you.

yeah thats why i was asking how big of a task it is, because i might be able to do it, no need to be angry bud

**Reply by ClixzyFN** [reply]
> > How hard it is doesn't matter, no one is going to do it for you.
> 
> yeah thats why i was asking how big of a task it is, because i might be able to do it, no need to be angry bud

it was only a question and he got mad for some reason

**Reply by Baker7** [reply]
@ClixzyFN  ... here is a place where you can download a fte android binary - there is an apk for download (I assume it works).

https://sourceforge.net/projects/fteqw/files/FTE%20Client/

**Reply by BrownNPC** [reply]
> @ClixzyFN ... here is a place where you can download a fte android binary - there is an apk for download (I assume it works).
> 
> https://sourceforge.net/projects/fteqw/files/FTE%20Client/

I went down this rabbit hole yesterday. This apk does not work, the last working version is from 2017, and that version won't run on Android greater than 9. 

I thought of a simpler solution though, we can take the web browser version and hook the game controller object. And make it so touch controls will modify that game controller object.

**Comment by S3KiR0** [comment]
Just use the browser version with a gamepad, that it'll do the job

**Reply by BrownNPC** [reply]
The browser version doesn't have PWA metadata so you cant remove the search bar while playing

**Comment by BrownNPC** [comment]
Good news: I got nzp booting on Android 
Bad news: we can't play the game cuz libs are missing 
![Screenshot_20241220-140657_Quad Touch](https://github.com/user-attachments/assets/5e4e16be-d8be-4b3d-8978-961167227198)

**Reply by BrownNPC** [reply]
I used an app called quadtouch, I downloaded the nzp PK3 files from GitHub ( the web version repo )
And put them in the Q1 folder for quadtouch 

They have an FTEQW port inside the app which let me at least boot the game

**Reply by BrownNPC** [reply]
Seems like libpng isn't linked in the app, also we can see which fte version they used, so I'm assuming we can replicate this outside of this app if we get that fte version's APK file

![Screenshot_20241220-141247_Quad Touch](https://github.com/user-attachments/assets/6416fc64-66b8-4730-a9f2-7b1f4388a360)

**Reply by BrownNPC** [reply]
[This](https://sourceforge.net/p/fteqw/code/5811/tree/) is the version being used in the app

If we find the nzp FTEQW fork version that uses this version and build that for Android, maybe just maybe  it will work

**Reply by Paindeluxe** [reply]
> [This](https://sourceforge.net/p/fteqw/code/5811/tree/) is the version being used in the app
> 
> If we find the nzp FTEQW fork version that uses this version and build that for Android, maybe just maybe it will work

Here's NZP's FTEQW Fork https://github.com/nzp-team/fteqw

**Reply by BrownNPC** [reply]
> > [This](https://sourceforge.net/p/fteqw/code/5811/tree/) is the version being used in the app
> > If we find the nzp FTEQW fork version that uses this version and build that for Android, maybe just maybe it will work
> 
> Here's NZP's FTEQW Fork https://github.com/nzp-team/fteqw

I know that, but we somehow gotta find the commit using this version of fteqw. ie. Version 5811

**Reply by BrownNPC** [reply]
Also gotta confirm this hypothesis by compiling the original fteqw 5811 to see if I'm correct about it working

**Comment by BrownNPC** [comment]
Ok I went into the app's discord and found their version of FTEQW. This version should be able to run the game 

https://github.com/emileb/fteqw/tree/mobile_dev

**Reply by Baker7** [reply]
In the event you are unable to get PNG working, you could convert all the .png textures to .tga. as a temporary workaround.

**Comment by BrownNPC** [comment]
![image](https://github.com/user-attachments/assets/953d8ceb-dc97-4ebd-80ab-cd780976779e)


# YOU CAN RUN NZP ON ANDROID
but there's a catch. You need to buy [this app](https://play.google.com/store/apps/details?id=com.opentouchgaming.quadtouch) 
Once you have the app, you will have to setup a folder on your android phone, the app guides you through it.
Download the game files. The easiest way is to just take the `nzp`folder from the pc version.

copy the contents of the nzp folder to  the-folder-you-setup-in-the-app/Q1/

then run the game.
You may need to use the console and type the map command to change the map.

**Reply by MotoLegacy** [reply]
This will only work in select instances, since the FTEQW build nzp uses is modified. Do not sign this off as a defacto way to play, thanks!

**Reply by m16a4666** [reply]
Greet from 2025,I recently wanted to try using quadtouch to run NZP, but I couldn't start the game. The steps are the same as you said. May I ask what version you are using to play? I am currently using the latest version of Windows and can't run it,check the picture below
![Screenshot_2025-08-17-23-56-46-43_b4de3044aa239eb1069d2839a72f95e2](https://github.com/user-attachments/assets/8984d45f-839c-4183-b376-bd747f5666c0)

**Reply by BrownNPC** [reply]
@m16a4666 i think you should join the quad touch discord. they have likely updated a few things.
I dont have the link and I dont remember how I got into it. but you can email the creator of the app, and ask him for an invite.
[emile.belanger@gmail.com](mailto:emile.belanger@gmail.com)

There also seems to be another app in town for free https://github.com/glKarin/com.n0n3m4.diii4a
I haven't tested it with nzp though. But it seems to support fteqw

**Comment by irandiaz** [comment]
Use the ppsspp emulator preferably gold version at least in my expe
![Screenshot_20250208_004308_org_ppsspp_ppssppgold_PpssppActivity](https://github.com/user-attachments/assets/c065b70a-3d68-4822-af5f-2daae4a47868)
rience it works well for me

**Comment by Mrhn2** [comment]
![1736193219714](https://github.com/user-attachments/assets/5a607f8f-0f2b-4903-a99e-8597f9c53c83)
No soy del equipo de nzp-team pero yo ya empeze hace un año a desarrollar una versión aparte soy MR HN desarrollo un port para Android con funciones nuevas como el cooperativo pantalla dividida, ya hace no mucho arregle el mapeo. Bueno el caso es que motolegacy tiene razón desarrollar una versión de Android es muy complejo aunque mi versión ya es jugable me a costado bastante.

**Reply by irandiaz** [reply]
Ya se puede descargar o todavía falta y si falta poco espero con emoción el pprt

**Reply by Mrhn2** [reply]
Mira mi canal de Youtube subo avances y todo soy @MRHN2008 en Youtube

**Comment by Mrhn2** [comment]
<img width="1600" height="720" alt="Screenshot_20250830-084747" src="https://github.com/user-attachments/assets/73c61a5d-b34e-48d4-83e2-9e4dcfffdf3e" />
Me perdí unos meses pero el proyecto ya está en etapas finales aún falta pulir varias cosas pero el port (remasterizado) para Android y también para PC ya casi está listo saldrá hasta noviembre por algunas cosas que estado ocupado 😅

**Reply by angelaldana429-star** [reply]
Oye bro, está listo el port? De verdad quiero intentar probarlo
